integration-tests/artifacts/installer/v1alpha1/installer.go

```diff
@@ -30,6 +30,13 @@ import (
 type DefaultInstaller struct {
 	object     *unstructured.Unstructured
 	predicates []Predicate
+	*unstruct.Kubeclient
+}
+
+// NewKubeClient returns a new instance of unstructured kubeclient
+func (i *DefaultInstaller) NewKubeClient(opts ...unstruct.KubeclientBuildOption) *DefaultInstaller {
```

Comment:
The method signature is confusing.
If its NewKubeClient(...) then it should return a new instance of kubeclient.

If the purpose was to create a new instance of DefaultInstaller, then it should have following:
```go
func NewDefaultInstaller(opts ...unstruct.KubeClientBuildOption) *DefaultInstaller {
  // ...
}
```

If we want to make use of Builder then code will look like following:
```go
func (b *Builder) WithKubeClient(opts ...unstruct.KubeClientBuildOption) *Builder {
  // ...
}
```